### PR TITLE
Cleans up pipe_network.dm and pipeline.dm

### PR DESCRIPTION
**Pull request by Laboredih123**
[ATMOSPHERICS][INTERNAL][CLEANLINESS]
## About the PR 
Moves the pipe files to the atmospherics module, dmdocs them, optimises them, and in general just cleans them up.

## Why's this needed? 
Slowness bad, ugliness bad, speed good, cleanliness good.

### CODE REVIEW

File: code/modules/atmospherics/pipe_network.dm [hunk 1]
```diff
@@ -1,49 +1,48 @@
...
-		for(var/obj/machinery/atmospherics/machine in normal_members)
-			machine.network_disposing(src)
-		normal_members.len = 0
-	normal_members = 0
-	if (line_members)
-		for(var/datum/pipeline/member in line_members)
-			member.network = null
-		line_members.len = 0
+
+	for(var/obj/machinery/atmospherics/machine as anything in normal_members)
```
Comment (ZeWaka) [inline code/modules/atmospherics/pipe_network.dm]: what happens when one of them blows up? you'll get a null in this list

Comment (Laboredih123) [reply code/modules/atmospherics/pipe_network.dm]: Tested, the exploding normal_member calls dispose on the pipe_network while still existing during disposing(), meaning there will never be a null in this list, hopefully.

Comment (ZeWaka) [reply code/modules/atmospherics/pipe_network.dm]: What about any of the other lists you are skipping the typechecking for?

Comment (Laboredih123) [reply code/modules/atmospherics/pipe_network.dm]: air_disposing_hook() is used by normal members for removing gases from /datum/pipe_network/var/list/gases, pipeline disposing removes its gases and itself from the pipe_network, and pipelines are disposed whenever a member disposes.

**Pull request merged**